research/acknownledge.txt

```diff
@@ -1 +1,47 @@
-Author: Chau Anh Nhat - 2352856
\ No newline at end of file
+The 2-dimensional cutting stock problem (2D CSP) is a classic optimization problem where the goal is to cut a large piece of material (e.g., metal, wood, or fabric) into smaller rectangular pieces to meet specific requirements while minimizing waste. It is commonly encountered in manufacturing and production industries where efficient material usage is essential for cost savings.
+
+Problem Definition
+Input Material: You start with a large rectangular sheet of a certain width and height, which serves as the stock from which smaller pieces are cut.
+
+Demanded Pieces: There’s a set of smaller rectangles, each with specific width and height, that must be cut from the stock. These pieces represent the demand and can vary in quantity and size. Each piece type has:
+
+A specific size (width and height).
+A required quantity.
+Goal: The objective is to cut the demanded pieces from the stock in a way that minimizes waste (unused material), or equivalently, maximizes the use of the stock sheet. Often, this includes:
+Minimizing the total number of stock sheets required.
+Minimizing the unused area on each stock sheet if a fixed number of sheets are used.
+
+Types of Constraints
+Several constraints typically govern the solution:
+- Stock Size Constraint: The cuts must fit within the dimensions of the stock sheet.
+- Demand Constraint: Each demanded piece must be produced in the required quantity.
+- Non-Overlap Constraint: The pieces cannot overlap on the stock sheet; each piece must occupy a unique position.
+- Cutting Pattern Constraint: Often, only certain types of cuts are allowed 
+  Types of Cutting Patterns
+  + Guillotine Cuts: Each cut must go from one edge of the material to the other, simplifying the cutting process. Guillotine cutting is often preferred because it's easier to automate and more practical in large-scale manufacturing.
+  + Non-Guillotine Cuts: Cuts don’t have to span the entire length or width, which can lead to more efficient use of material but is usually harder to implement due to increased cutting complexity.
+
+Example Problem
+Imagine a factory is cutting a large metal sheet that is 80 inches by 50 inches. The factory needs to produce a range of parts with specific sizes to minimize waste while meeting various order requirements. The parts required are:
+
+Six pieces of size 15x10 inches
+Five pieces of size 10x10 inches
+Four pieces of size 12x15 inches
+Eight pieces of size 6x8 inches
+Ten pieces of size 5x5 inches
+
+Difficulties in Solving this Example
+- Piece Size and Shape Diversity:
+  + The pieces have different dimensions, some long and narrow, others square or almost square. This variety complicates the arrangement, as pieces cannot       easily “tessellate” or fit together like a puzzle without leaving gaps.
+  + Large pieces, such as 20x10, reduce flexibility in the arrangement, limiting the space left for smaller pieces.
```

Comment:
Typing error

Reply:
fixed, changed from 20x10 to 15x10, if there's any other error let me know